compatibility_server/pip_checker.py

```diff
@@ -27,29 +27,48 @@
 import enum
 import json
 import logging
-import os.path
 import shlex
-import subprocess
-import tempfile
+import socket
 import urllib.request
 
 from typing import Any, List, Mapping, Optional
 
+import docker
+
+socket.setdefaulttimeout(3)
+
 PYPI_URL = 'https://pypi.org/pypi/'
+CONTAINER_WITH_PKG = "checker"
+TIME_OUT = 300  # seconds
+
+
+class PipCheckerError(Exception):
+    """Pip checker failed in an unexpected way."""
 
+    def __init__(self, error_msg: str):
+        super(PipCheckerError, self).__init__(
+            'Pip Checker failed, '
+            'logs are: {error_msg}'.format(error_msg=error_msg))
 
-class PipError(Exception):
+        self.error_msg = error_msg
+
+
+class PipError(PipCheckerError):
     """A pip command failed in an unexpected way."""
 
-    def __init__(self, command: List[str], returncode: int, stderr_path: str):
+    def __init__(self,
+                 error_msg: List[str],
+                 command: List[str],
+                 returncode: int):
         super(PipError, self).__init__(
-            'pip command ({command}) failed with error [{returncode}], '
-            'logs at: {stderr_path}'.format(
+            'Pip command ({command}) failed with error [{returncode}], '
+            'logs at: {error_msg}'.format(
                 command=command, returncode=returncode,
-                stderr_path=stderr_path))
+                error_msg=error_msg))
+
+        self.error_msg = error_msg
```

Comment:
I think that the base class will do this.

Reply:
Removed.